src/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+
+import styles from './Filter.module.scss';
+
+import { FilterFormTypes } from '../../types/FilterFormTypes';
+import { FilterTypes } from '../../types/FilterTypes';
+import { UserInfo } from '../../types/UserInfo';
+
+export const Filter: React.FC<FilterTypes> = ({ list, setFilterUsers }) => {
+  const { register, handleSubmit, reset } = useForm<FilterFormTypes>({
+    mode: 'onBlur',
+    defaultValues: {},
+  });
+
+  const onSubmit = (data: FilterFormTypes) => {
```

Comment:
will be moved

Reply:
list = list ?. filter ((users) => !data.firstName || item.firstName.toLowerCase().includes(data.firstName))
?.filter(... age from )
?.filter(... age to  )
?.filter(... gender) ?? []